planner/core/point_get_plan.go

```diff
@@ -186,19 +197,23 @@ func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt) *PointGetP
 	if pairs == nil {
 		return nil
 	}
-	handlePair, unsigned := findPKHandle(tbl, pairs)
+	handlePair, fieldType := findPKHandle(tbl, pairs)
 	if handlePair.value.Kind() != types.KindNull && len(pairs) == 1 {
 		schema := buildSchemaFromFields(ctx, tblName.Schema, tbl, selStmt.Fields.Fields)
 		if schema == nil {
 			return nil
 		}
 		p := newPointGetPlan(ctx, schema, tbl)
-		var err error
-		p.Handle, err = handlePair.value.ToInt64(ctx.GetSessionVars().StmtCtx)
+		intDatum, err := handlePair.value.ConvertTo(ctx.GetSessionVars().StmtCtx, fieldType)
 		if err != nil {
+			if terror.ErrorEqual(types.ErrOverflow, err) {
+				p.IsTableDual = true
+				return p
+			}
 			return nil
 		}
-		p.UnsignedHandle = unsigned
+		p.Handle = intDatum.GetInt64()
+		p.UnsignedHandle = mysql.HasUnsignedFlag(fieldType.Flag)
```

Comment:
`fieldType` can be nil pointer according to function `fieldType()`

Reply:
`fieldType` is the FieldType of PKHandle, it's can not be nil theoretically. https://github.com/pingcap/tidb/pull/10676/files#diff-f888b92d66ee2d478ea975b08cddaa6cR200